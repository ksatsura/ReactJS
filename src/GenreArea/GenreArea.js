//imports
import React from 'react';
import '../style.css';

export const GenreArea = (props) => {

    return <p className='asset-genre'>Films by {props.genre} genre</p>;
};

export default GenreArea;