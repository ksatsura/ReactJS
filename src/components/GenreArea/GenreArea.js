//imports
import React from 'react';
import '../../style.css';

// @flow
type Props = {
    genre: string
};

export const GenreArea = (props) => {

    return <p className='film-genre'>Films by {props.genre} genre</p>;
};

export default GenreArea;