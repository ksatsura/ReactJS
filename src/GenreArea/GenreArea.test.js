import React from 'react';
import { GenreArea } from './GenreArea';
import { shallow } from 'enzyme';

describe('GenreArea.js', () => {

    const props = {
        genre: '',
    };

    describe('renders', () => {

        it('renders for desktop', () => {
            const component = shallow(<GenreArea {...props} />);

            expect(component).toMatchSnapshot();
        });

        it('renders for mobile', () => {
            const component = shallow(<GenreArea {...props} isMobile />);

            expect(component).toMatchSnapshot();
        });
    });
});