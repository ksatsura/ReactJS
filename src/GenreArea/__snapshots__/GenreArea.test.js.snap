// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreArea.js should render as expected 1`] = `
<p
  className="film-genre"
>
  Films by 
   genre
</p>
`;
